recipes/nanorange/all/conanfile.py

```diff
@@ -0,0 +1,33 @@
+import os
+
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class NanorangeConan(ConanFile):
+    name = "nanorange"
+    version = "20191001"
+    license = "Boost 1.0"
+    author = "Paul M. Bendixen paulbendixen@gmail.com"
+    url = "github.com/conan-io/conan-center-index"
+    website = "https://github.com/tcbrindle/NanoRange"
+    description = "NanoRange is a C++17 implementation of the C++20 Ranges proposals (formerly the Ranges TS)."
+    topics = ("ranges", "C++17", "Ranges TS")
+    no_copy_source = True
+    settings = "compiler"
+    # No settings/options are necessary, this is header only
+
+    def configure(self):
+        if not any([str(self.settings.compiler.cppstd) == std for std in ["17", "20", "gnu17", "gnu20"]]):
```

Comment:
```suggestion
        if self.compiler.cppstd:
           if not any([str(self.settings.compiler.cppstd) == std for std in ["17", "20", "gnu17", "gnu20"]]):
               raise ConanInvalidConfiguration("Current cppstd is lower than required c++ standard.")
   
       # detect supported cxx standard according the current compiler
       elif not cppstd_flag(self.settings.compiler,         self.settings.compiler.version, "17"):
           raise ConanInvalidConfiguration("Current compiler doesn't not support the required c++ standard.")
```
By default, cppstd is None, which doesn't fit your rule, as you need C++17 at least, you also need to verify the compiler version, since you don't have cppstd all the time.
You need to import `from conans.client.build.cppstd_flags import cppstd_flag`

Reply:
that one is not a public tool, we'd rather export it as a tool in the `tools` module in order to use it or find another approach

Reply:
So the request can't get merged until the tools module is updated ?

Reply:
@uilianries if that tool is interesting we can discuss in a Conan issue the possibility of making it public. Would you mind opening an issue referencing this case?

Reply:
@danimtb https://github.com/conan-io/conan/issues/5958, also I've opened a PR with such solution https://github.com/conan-io/conan/pull/5997

Reply:
Turns out, to be really nice to the Visual Studio users, I should also check that the version is at least 15.9
Latest push opens up to this being a posibility.